### ISSUE TITLE
Empty space at the bottom of the screen on iOS with Cordova WKWebview

### ISSUE THREAD

**Issue by louisameline** [quasarframework/quasar#6695]
**Describe the bug**

The bug is the white space at the bottom of the screenshot below. The bottom-sheet and its dark underlay should reach the bottom of the screen.

![image](https://user-images.githubusercontent.com/371814/77784089-3c582a80-705a-11ea-94ca-ea72e93b27d2.png)

**Remarks**
~~This happens because of this property:~~ 
```
.q-body--prevent-scroll
  .q-layout
    position: fixed
```
~~Which was introduced because of this: https://github.com/quasarframework/quasar/issues/4497
It's solved as soon as I disable `position: fixed`. I fiddled with "height: 100%", "bottom: 0" etc. but couldn't find a way to work around this wkwebview `position: fixed` bug.~~
~~I suggest to try an alternate way to prevent scrolling, as described on https://stackoverflow.com/questions/3047337/does-overflowhidden-applied-to-body-work-on-iphone-safari~~
Tested on a real iPhone XS, with iOS 13.3.1
Happens with or without any cordova plugins installed.
It does not happen when using Capacitor, or Android, or in a browser.

Note: strangely, the bug no longer happens after the device is rotated at least once.

**Comment by louisameline** [comment]
I think the behavior of that bug has changed a little in iOS 13.4. Removing `position: fixed` no longer solves the issue. Also, I saw that the issue is not limited to the times when `.q-body--prevent-scroll` is used.

I'm not sure but the change of behavior might be a consequence of work done on a very similar issue where space was left at the bottom after using the keyboard https://github.com/apache/cordova-ios/issues/417

I discovered the guys at Ionic talk about this as well:
https://github.com/ionic-team/ionic/issues/12948

And this issue here also talk about it:
https://github.com/apache/cordova-plugin-statusbar/pull/87

Seems somewhat related to `viewport-fit=cover`.

In my tests, it seems to happen every time the page height is not as big as the screen size.
Basically, Quasar sets a min-height in the style tag of the q-layout div, and this min-height is actually smaller than the screen size, so that's why there is space left at the bottom of the page. So I guess there must be a bug in how the screen size is detected in iOS + wkwebview, maybe there is a possible temporary fix to be found here.

I'm still trying a couple things at this time. This bug is a mess.

**Comment by louisameline** [comment]
It looks like the Ionic folks wrote a patch for this. You need to:
- remove cordova's Wkwebview plugin from your project, including cleaning your config.xml from the related entries
- install the ionic webview (which includes the Wkwebview plugin + their own improvements):
`cordova plugin add cordova-plugin-ionic-webview --save`
- delete the `ios` folder in the `platforms` folder and rebuild

More info at https://github.com/ionic-team/cordova-plugin-ionic-webview

**Comment by louisameline** [comment]
I just found that the Ionic webview is currently not compatible with Firebase's SignInWithRedirect.
See https://github.com/firebase/firebase-js-sdk/issues/1244 and https://github.com/firebase/firebaseui-web/issues/365

**Comment by louisameline** [comment]
I've reverted to Cordova's WKWebView. After more investigation, it turns out that the `window.innerHeight` returned by the webview is at some occasions the window height MINUS the safe area inset height. It's an issue that has been reported here: https://github.com/apache/cordova-plugin-wkwebview-engine/issues/108

I fiddle around all day and found a temporary fix. Add this to the `mounted` hook of App.vue:
```javascript
const qAppStyle = document.querySelector('#q-app').style
qAppStyle.minHeight = '1000px'

// a nextTick doesn't cut it here
setTimeout(() => {
	document.body.style.display = 'contents'
	qAppStyle.minHeight = ''
}, 0)
```
Caveat due to `display: contents;`: the padding and margin of the body, if any, will not be enforced anymore, you'll have to set it on children instead. Also, a background set on the body will not work anymore, but you can set it on the `html` tag instead.

**Comment by louisameline** [comment]
Better fix than my previous message: add this at the beginning of the body in `index.template.html`

```html
<div style="height: 100vh; position: absolute; visibility: hidden; width: 1px;"></div>
```
This will not work whenever the body has `position: fixed` though, so I had to override `.q-body--prevent-scroll` like this:
```javascript
.q-body--prevent-scroll.platform-ios {
	overflow: hidden;
	position: relative !important;
}
```

**Comment by rstoenescu** [comment]
We do recommend the Ionic Webview.
Unfortunately, we did everything possible from within Quasar. If there is something that further needs to be fixed, it can only be done by Apple or by the Ionic folks with their webview.

**Comment by jukrb0x** [comment]
That's how I fix this problem.

1. Find out where is the scrolling bar, for me, it's `q-layout`
2. Add new class `scrolling-trouble` to `q-layout`
3. Add the following snippet to style
```scss
.scrolling-trouble {
  overflow-y: hidden !important;
  height: 1px;
}
```

Inspired by & Thanks @FlickerSoul

**Comment by SameerInstacash** [comment]
ionic cordova plugin rm cordova-plugin-ionic-webview
ionic cordova plugin add cordova-plugin-ionic-webview@latest

Try this, I fixed same issue using above link

**Comment by sylvinrodz** [comment]
**viewport-fit=cover** just add this to your meta tag 
`<meta name="viewport" content="width=device-width, initial-scale=1"/>`
